org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/util/JRTUtil.java

```diff
@@ -144,11 +119,9 @@ public static JrtFileSystem getJrtSystem(File image) {
 	 */
 	public static JrtFileSystem getJrtSystem(File image, String release) throws IOException {
 		Jdk jdk = new Jdk(image);
-		String key = jdk.path;
-
-		if (release != null && !jdk.sameRelease(release)) {
-			key = key + "|" + release; //$NON-NLS-1$
-		}
+		String key = release != null && !jdk.sameRelease(release) //
```

Comment:
For me the original version is easier to read, and the new ternary expression is harder. I would keep it "as is"